### PR TITLE
Feat/omega 34 user settings menu

### PR DESCRIPTION
**Pull request by eTallang**
OMEGA-34

## What is the reason for this change?
To move the links found on the "Min profil" to the user menu in the top right. This is a part of the redesign of "Min side".

### What is the expected result?
A new user menu that has more content than the user menu in prod today.

## Any changes/updates that may affect others?
|                                     |Changed|
|-------------------------------------|-------|
| Unit tests                          |      |
| Dependencies                        |       |
| Environment variables               |       |
| Build scripts                       |       |

## How can it be tested and verified?

- [ ] Check that the new user menu looks like the designs in [Figma](https://www.figma.com/file/sJHoScmQRq0hlr6AcHI9WB/Elvia.no?node-id=10104%3A63803).
- [ ] Check that the links in the menu goes to the right locaions.

### CODE REVIEW

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -1,27 +1,31 @@
 OMEGA-X
 
 ## What is the reason for this change?
-Titles use incorrect font styles.  
+
```
Comment (eTallang) [inline docs/pull_request_template.md:4]: I'm really not sure why this became a part of the PR. I've never changed this, and I'm not even allowed to commit a revert :confused:

File: packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss [hunk 2]
```diff
@@ -0,0 +1,79 @@
+@import '../../theme';
+
+.my-page-link {
+  display: flex;
+  align-items: center;
+  text-decoration: none;
+  color: map-get($colors, 'black');
+  font-weight: 700;
+  font-size: emCalc(20px);
+  margin-bottom: 2.5rem;
```
Comment (FredricW) [inline packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss:10]:
```suggestion
  margin-bottom: spaceCalc(2.5);
```

File: packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  white-space: nowrap;
+  letter-spacing: 0;
+}
+
+.my-page-link-icon {
+  margin-left: 8px;
```
Comment (FredricW) [inline packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss:16]: spaceCalc
```suggestion
  margin-left: spaceCalc(1);
```

File: packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+.menu-dropdown {
+  z-index: z('base');
+  background: map-get($colors, 'white');
+  box-shadow: 0 0 40px 0 map-get($colors, 'shadow6');
+  top: 73px;
+  right: 1rem;
+  @include borderRadiusMedium;
+  padding: 24px;
+  position: absolute;
+  min-width: 270px;
```
Comment (FredricW) [inline packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss:28]: Use spaceCalc and emCalc instead of setting px values directly 8)

**Review by FredricW — CHANGES_REQUESTED** [review]
I understand that some of the code is just moved, but it would be nice if you can tidy up the sass a bit! Also some comments on the typography, otherwise looking good.

File: packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss (hunk 2 above)
Comment (lobunto) [reply packages/web/src/components/HeaderUserMenu/HeaderUserMenu.module.scss:10]: Isn't `spaceCalc(5)` more correct here @FredricW?

**Comment by OysteinThuen** [comment]
Build #3068 successfully built and deployed.

- Prod: https://5f4e311a529e1ad2d784b3f1--elvia-prod.netlify.app
- Test: https://5f4e314067e48bc899784338--elvia-test.netlify.app

**Comment by FredricW** [comment]
Does it make sense that "Logg ut fra andre enheter" is placed in "Inlogging" and not in "Logg ut" ?

**Comment by eTallang** [comment]
> Does it make sense that "Logg ut fra andre enheter" is placed in "Inlogging" and not in "Logg ut" ?

Checking with Malin.

**Comment by eTallang** [comment]
> Does it make sense that "Logg ut fra andre enheter" is placed in "Inlogging" and not in "Logg ut" ?

@FredricW We agreed with your suggestion, so the link has been moved. Ready for another review.

File: packages/web/src/components/HeaderUserMenu/HeaderUserMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      )}
+
+      <TextCaps className={css.menuDropdownTitle}>
+        {getTerm(Terms.minSideProfileSignInTitle)}
+      </TextCaps>
+      {signInLinks.map((link, index) => (
+        <a className={css.menuLink} href={createHidUrl(link.url)} key={index}>
+          <div className={css.menuLinkIcon}>{getIcon(link)}</div>
+          {getTerm(link.name)}
+        </a>
```
Comment (lobunto) [inline packages/web/src/components/HeaderUserMenu/HeaderUserMenu.tsx:118]: Links related to customer relationship will probably not link to Elvid but be relative to mypage. `createHidUrl` should not create links for customer relationships

**Review by lobunto — CHANGES_REQUESTED** [review]

File: packages/web/src/components/HeaderUserMenu/HeaderUserMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+      <TextCaps className={css.menuDropdownTitle}>
+        {getTerm(Terms.minSideProfileCustomerRelationshipTitle)}
+      </TextCaps>
+      {customerRelationshipLinks.map((link, index) => (
+        <a className={css.menuLink} href={createHidUrl(link.url)} key={index}>
+          <div className={css.menuLinkIcon}>{getIcon(link)}</div>
+          {getTerm(link.name)}
+        </a>
+      ))}
```
Comment (lobunto) [inline packages/web/src/components/HeaderUserMenu/HeaderUserMenu.tsx:129]: Links related to customer relationship will probably not link to Elvid but be relative to mypage. `createHidUrl` should not create links for customer relationships

**Review by lobunto — CHANGES_REQUESTED** [review]

File: packages/web/src/components/HeaderUserMenu/HeaderUserMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+export const HeaderUserMenu: React.FC<HeaderUserMenuProps> = ({
+  logOutRef,
+  minSideLink,
+  minSideNavigation,
+}) => {
+  const userHasEmail = pipe(prop('email'), isNil, not)
+  const userProfile = useSelector(oidcSelectors.getUserProfile)
+  const redirectUri = window.location
+  const createHidUrl = (pathToHid: string) =>
```
Comment (lobunto) [inline packages/web/src/components/HeaderUserMenu/HeaderUserMenu.tsx:37]: *Hid* should also probably be renamed to *Elvid*

**Comment by lobunto** [comment]
I cannot tell why, but it seems that an illustration is gone from this PR..? In production it looks fine.

Screenshot 2020-09-02 at 13 19 26

**Comment by eTallang** [comment]
@lobunto Problem solved. Ready for new review

**Comment by lobunto** [comment]
> @lobunto Problem solved. Ready for new review

Great, it looks really good :) I've got one more issue. When the link "Be om tilgang" in the menu is clicked the menu doesn't hide. And in some cases, it just quickly hides before it becomes visible again.

**Review by FredricW — APPROVED** [review]
Once Marius' comment is addressed I think it is good to go

**Comment by eTallang** [comment]
> > @lobunto Problem solved. Ready for new review
> 
> Great @eTallang , it looks really good :) I've got one more issue. When the link "Be om tilgang" in the menu is clicked the menu doesn't hide. And in some cases, it just quickly hides before it becomes visible again.

Fixed :+1:

**Review by lobunto — APPROVED** [review]

**Pull request merged**